deploy/runtools/run_farm_deploy_managers.py

```diff
@@ -656,6 +661,37 @@ def clear_fpgas(self) -> None:
                 with StreamLogger('stdout'), StreamLogger('stderr'):
                     run(f"xbutil reset -d {card_bdf} --force")
 
+    def localize_xclbin(self, slotno: int) -> None:
+        """ download xclbin URI to remote node. """
+        assert slotno < len(self.parent_node.sim_slots)
+        serv = self.parent_node.sim_slots[slotno]
+        hwcfg = serv.get_resolved_server_hardware_config()
+        if re.match(_RFC_3986_PATTERN, hwcfg.xclbin):
+            remote_home_dir = self.parent_node.get_sim_dir()
+            remote_sim_dir = """{}/sim_slot_{}/""".format(remote_home_dir, slotno)
+            hwcfg.local_xclbin = './local.xclbin'
+
+            def get_xclbin(xclbin_uri:str, xclbin_lpath:PathLike) -> None:
+                # TODO instead of blowing up if the file exists, consider using fsspec
+                # filecache https://filesystem-spec.readthedocs.io/en/latest/features.html#caching-files-locally
+                # so that multiple slots using the same xclbin only grab it once and
+                # we only download it if it has changed at the source.
+                # HOWEVER, 'filecache' isn't thread/process safe and I'm not sure whether
+                # this runs in @parallel for fabric
```

Comment:
This code should run in parallel on multiple run hosts but once you are using a specific run host this is done serially: 

https://github.com/firesim/firesim/blob/7839d4fda5eb360841694833c40f50e3e9bf3387/deploy/runtools/firesim_topology_with_passes.py#L388-L393

https://github.com/firesim/firesim/blob/7839d4fda5eb360841694833c40f50e3e9bf3387/deploy/runtools/run_farm_deploy_managers.py#L598

Reply:
I think using the file cache should be fine (and doesn't need a lock)

Reply:
Ok.  It's good to know that it should work now.  What happens if someone changes the code so that a lock is needed?  How do we detect that case and blow up?

Said differently, can we test to ensure that it should continue to be okay?

Reply:
I vote to just have a comment indicating that the code should not be run in parallel on the same host. I think because F1 requires that we do things serially, we will just enforce this in the future.

Reply:
We decided to punt on FileCache for this PR.